hooks/log_metrics.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""Hook that gets executed every time Toolkit logs a user metric."""
+
+from tank import Hook
+ 
+
+class LogMetrics(Hook):
+    
+    def execute(self, tk, metrics):
+        """Called when Toolkit logs a user metric.
+        
+        :param tk: Toolkit instance 
+        :param list metrics: list of `ToolkitMetric` objects that were logged.
```

Comment:
I think `ToolkitMetric` should be a class internal to core. Could we just serialize the data out as a dictionary?